sflib.py

```diff
@@ -1299,106 +1302,45 @@ def googleIterate(self, searchString, opts=dict()):
-                failed = False
-            else:
-                break
+        endpoint = "https://api.cognitive.microsoft.com/bing/v7.0/search?"
+        params = {
+                "q": searchString,
+                "responseFilter": "Webpages",
+                "count": opts["count"],
+            }
+        response = self.fetchUrl(
+            endpoint + urllib.urlencode(params),
+            timeout=opts["timeout"],
+            useragent=opts["useragent"],
+            headers={"Ocp-Apim-Subscription-Key": opts["api_key"]},
+        )
 
-        if attempts == 3:
+        if response['status'] != 'OK':
+            self.error("Failed to get a valid response from bing API", exception=False)
             return None
 
-        returnResults[seedUrl] = firstPage['content']
-        pat = re.compile("(\/search\S+first=\d+.[^\'\"]*FORM=\S+)", re.IGNORECASE)
-        matches = re.findall(pat, firstPage['content'])
-        while matches > 0 and fetches < limit:
-            nextUrl = None
-            fetches += 1
-            for match in matches:
-                # Bing moves in increments of 10
-                if "first=" + str((fetches * 10) + 1) in match:
-                    nextUrl = match.replace("&amp;", "&").replace("%3a", ":")
-
-            if nextUrl is None:
-                self.debug("Nothing left to scan for in Bing results.")
-                return returnResults
-            self.info("Next Bing URL: " + nextUrl)
-
-            # Wait for a random number of seconds between fetches
-            if 'nopause' not in opts:
-                pauseSecs = random.randint(4, 15)
-                self.info("Pausing for " + str(pauseSecs))
-                time.sleep(pauseSecs)
-
-            attempts = 0
-            while attempts < 3:
-                nextPage = self.fetchUrl(u'https://www.bing.com' + nextUrl,
-                                         timeout=opts['timeout'], useragent=opts['useragent'])
-                if nextPage['code'] == "400":
-                    self.error("Bing doesn't like us any more..", False)
-                    failed = True
+        response_json = json.loads(response['content'])
 
-                if nextPage['content'] is None:
-                    self.error("Failed to fetch subsequent content from Bing.", False)
-                    failed = True
-                else:
-                    if "/challengepic?" in firstPage['content']:
-                        self.error("Bing returned a CAPTCHA.", False)
-                        failed = True
-
-                if failed:
-                    self.refreshTorIdent()
-                    attempts += 1
-                    failed = False
-                else:
-                    break
+        if "webPages" in response_json:
+            results = {
+                "urls": [result["url"] for result in response_json["webPages"]["value"]],
+                "webSearchUrl": response_json["webPages"]["webSearchUrl"],
+            }
```

Comment:
The above two lines have assumptions that the sub-keys will exist. This will break badly if it ever changes and should be handled more cleanly.